forward/fwd.go

```diff
@@ -190,18 +210,6 @@ func (f *httpForwarder) serveHTTP(w http.ResponseWriter, req *http.Request, ctx
 		ctx.log.Infof("Round trip: %v, code: %v, duration: %v",
 			req.URL, response.StatusCode, time.Now().UTC().Sub(start))
 	}
-
-	defer response.Body.Close()
-
-	if err != nil {
-		ctx.log.Errorf("Error copying upstream response Body: %v", err)
-		ctx.errHandler.ServeHTTP(w, req, err)
-		return
-	}
-
-	if written != 0 {
```

Comment:
It's a set of a header after this initial WriteStatus, which isn't allowed.  The only way to do this would be to buffer the entire response, write the content-length, then write the body out. You have to assume that the original handler has already set the content-length. It's unrelated to the actual fix to trailers, but it's definitely not a valid thing to do.